mockClient.go

```diff
@@ -207,7 +208,7 @@ func (mock mockClient) ExpectationsWereMet() error {
 func (mock *mockClient) Expect(method string, path string) *MockRequest {
 	if mock.next > 0 {
 		msg := "requests have already been made"
-		panic(fmt.Errorf("%s: %w: %s", mock.name, ErrCannotChangeExpectations, msg))
+		panic(fmt.Errorf("%s: %w: %s", mock.id, ErrCannotChangeExpectations, msg))
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/blugnu/http/pull/3#issuecomment-2211525124>#3867f3</a><span> - 07/06/2024, 12:12 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The panic message uses 'mock.id' which is a change from 'mock.name'. If 'mock.id' is not properly initialized or if it is an empty string, the error message might not be informative enough to debug the issue.
</div>
<div id="fix">
<b>Fix:</b> Ensure that 'mock.id' is properly initialized and contains a meaningful value before it is used in the panic message. If 'mock.id' can be empty or uninitialized, consider adding a check or a default value to make the error message more informative.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -209,6 +209,10 @@
  func (mock *mockClient) Expect(method string, path string) *MockRequest {
      if mock.next > 0 {
          msg := "requests have already been made"
 +        if mock.id == "" {
 +            mock.id = "unknown-client"
 +        }
          panic(fmt.Errorf("%s: %w: %s", mock.id, ErrCannotChangeExpectations, msg))
      }
```
</div>
</details>
</div>
<br>


</div>